cluster_ha_resources.go

```diff
@@ -0,0 +1,81 @@
+package proxmox
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+func (cl *Cluster) CreateHAResource(ctx context.Context, haResource HAResource) error {
+	if err := cl.client.Post(ctx, "/cluster/ha/resources", &haResource, nil); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (cl *Cluster) DeleteHAResource(ctx context.Context, sid SID) error {
+	if err := cl.client.Delete(ctx, fmt.Sprintf("/cluster/ha/resources/%s:%d", sid.Type, sid.ID), nil); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (cl *Cluster) GetHAResource(ctx context.Context, sid SID) (HAResource, error) {
+	var haResource HAResource
+	if err := cl.client.Get(ctx, fmt.Sprintf("/cluster/ha/resources/%s:%d", sid.Type, sid.ID), &haResource); err != nil {
+		return haResource, err
+	}
+
+	return haResource, nil
+}
+
+func (cl *Cluster) ListHAResources(ctx context.Context, resourceType HAResourceType) ([]HAResource, error) {
+	var haResources []haResource
+	if err := cl.client.Get(ctx, fmt.Sprintf("/cluster/ha/resources?type=%s", resourceType), &haResources); err != nil {
+		return nil, err
+	}
+
+	return fillHAResources(haResources)
+}
+
+func fillHAResources(resources []haResource) ([]HAResource, error) {
+	haResources := make([]HAResource, 0, len(resources))
+	for _, resource := range resources {
+		sid, err := parseSid(resource.Sid)
+		if err != nil {
+			return nil, fmt.Errorf("parse sid: %w", err)
+		}
+
+		haResources = append(haResources, HAResource{
+			ID:          sid.ID,
+			Group:       AsPtr(resource.Group),
+			Comment:     AsPtr(resource.Comment),
+			MaxRelocate: AsPtr(resource.MaxRelocate),
+			MaxRestart:  AsPtr(resource.MaxRestart),
+			State:       AsPtr(resource.State),
+		})
+	}
+
+	return haResources, nil
+}
+
+func parseSid(sid string) (SID, error) {
```

Comment:
could you call this parseSID and wrap a unit test around it so no one breaks in the future accidently? something simple with some sample strings to test each conditional